untyped-lambda-calculus/article.md

```diff
@@ -146,63 +146,73 @@ This brings us to the discussion of evaluation strategies.
+In other words, the applicative order strategy first reduces all of the arguments (from left to right) and then applies the abstraction, while normal order does the inverse.
 
-These two strategies are free to go inside abstractions to find redexes.
-We call them _strong_.
+#### Reducing abstraction bodies or not
 
-Since we have strong strategies, it's reasonable to expect we also have weak ones.
-Weak strategies can't go inside abstraction bodies.
-The weak strategy corresponding to the normal order is called _call-by-name_, and the one corresponding to the applicative order is _call-by-value_.
+The two strategies above are free to go inside abstractions to find redexes.
+For this reason, we call them _strong_.
 
-When we're talking about programming languages, we can't necessarily inspect function bodies, so the weak strategies are considered more practical.
+But there are also strategies that can't go inside abstraction bodies, which we call _weak_.
+When we're talking about programming languages, we can't necessarily inspect function bodies, so the weak strategies are considered more practical there.
 
-It's interesting to note that the call-by-value strategy corresponds to eager evaluation.
+The weak strategy corresponding to the normal order is called _call-by-name_, and the one corresponding to the applicative order is _call-by-value_.
+Call-by-value corresponds to eager evaluation.
 Call-by-name, on the other hand, corresponds to lazy evaluation.
+
 In eager evaluation, arguments are always evaluated before the function call.
 Most programming languages follow this strategy by default.
 In lazy evaluation, some arguments are potentially never evaluated at all, but some computations may be duplicated.
 Very few programming languages implement lazy evaluation this way (the only real example that comes to mind is Algol 60).
 
-A practically significant variation on call-by-name is _call-by-need_, which is, at least in theory, used in Haskell.
+A more practical variation of call-by-name is _call-by-need_, which is, at least in theory, used in Haskell.
 Call-by-need behaves exactly like call-by-name, except that reduction steps that would duplicate computations, don't.
-The trick is to share the computation corresponding to an argument everywhere it appears, such that the shared computation is performed at most once.
+It does that by sharing the computation corresponding to an argument everywhere it appears, so any shared computation is performed at most once.
 
-Expressions that are not reducible using the chosen evaluation strategy are said to be in the _normal form_.
+To sum up, here's a table with all the mentioned strategies and their characteristics:
 
-If a given term has a normal form under both strong strategies, it's the same for both.
-Note that applicative order strategy is strictly weaker than normal order strategy.
-Some terms that normalize under normal order don't under applicative order.
```

Comment:
> Some terms that normalize under normal order don't under applicative order.

This is a pretty important point, actually. It follows from the rest of the article, but I think it's good to spell it out.

At least, it would be good to mention that normal order/call-by-name are respectively strongly/weakly normalizing, whereas other strategies aren't necessarily.